src/gateway/router/routes/contractsRoute.ts

```diff
@@ -6,63 +6,58 @@ const MAX_CONTRACTS_PER_PAGE = 100;
+  contractLimit && bindings.push(contractLimit);
+  interactionLimit && bindings.push(interactionLimit);
 
   try {
     const benchmark = Benchmark.measure();
     const result: any = await gatewayDb.raw(
       `
-          SELECT c.contract_id                                                   AS contract,
-                 c.owner                                                         AS owner,
-                 c.type                                                          AS contract_type,
-                 c.pst_ticker                                                    AS pst_ticker,
-                 c.pst_name                                                      AS pst_name,
-                 c.testnet                                                       AS testnet,
-                 s.src_content_type                                              AS src_content_type,
-                 s.src_wasm_lang                                                 AS src_wasm_lang,
-                 count(i.contract_id)                                            AS interactions,
-                 count(case when i.confirmation_status = 'corrupted' then 1 end) AS corrupted,
-                 count(case when i.confirmation_status = 'confirmed' then 1 end) AS confirmed,
-                 max(i.block_height)                                             AS last_interaction_height,
-                 count(*) OVER ()                                                AS total
-          FROM contracts c
-                   LEFT JOIN interactions i
-                             ON c.contract_id = i.contract_id
-                   LEFT JOIN contracts_src s
-                             ON c.src_tx_id = s.src_tx_id
-          WHERE c.contract_id != ''
-            AND c.type != 'error' 
-                ${contractType ? ' AND c.type = ?' : ''} 
-                ${sourceType ? ' AND s.src_content_type = ?' : ''}
-                ${testnet ? ' AND c.testnet IS NOT NULL' : ''}
-          GROUP BY c.contract_id, c.owner, c.type, c.pst_ticker, c.pst_name, s.src_content_type, s.src_wasm_lang
-          ORDER BY last_interaction_height DESC NULLS LAST, interactions DESC NULLS LAST ${
-            parsedPage ? ' LIMIT ? OFFSET ?' : ''
-          };
+        with contract as (select 'contract'      AS contract_or_interaction,
+                                 contract_id     AS contract,
+                                 ''              AS interaction_id,
+                                 owner           AS owner,
+                                 type            AS contract_type,
+                                 ''              AS function,
+                                 block_height    AS block_height,
+                                 deployment_type AS source
+                          from contracts
+                          where contract_id != ''
+                            AND type != 'error' 
+        ${testnet ? ' AND c.testnet IS NOT NULL' : ''}
```

Comment:
and what about the mainnet? i guess right now testnet txs are included in the mainnet query...

Reply:
Ok, I changed it so that either mainnet or testnet entries are shown, never both.